### PR TITLE
ROX-24511: Add check details information tab

### PR DESCRIPTION
**Pull request by bradr5**
## Description

This adds a "Details" tab to the check details page with additional information about the current check.

Notes:

- Since the check details call is a separate request from the check details results, it made more sense to add the information to a details tab instead of including the same information in each row of the table.
- Clicking on a check inside Cluster Details now takes you to that check's details tab.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Screenshot 2024-06-11 at 11 52 24 AM

### CODE REVIEW

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ClusterDetailsTable.tsx [hunk 1]
```diff
@@ -146,10 +139,10 @@ function ClusterDetailsTable({
                                     <Tr key={checkName}>
                                         <Td dataLabel="Check">
                                             <Link
-                                                to={generatePath(coverageCheckDetailsPath, {
+                                                to={`${generatePath(coverageCheckDetailsPath, {
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ClusterDetailsTable.tsx:142]: Although not a blocker to merge, let’s start to replace `generatePath` with the following:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/utils/urlUtils.ts#L40-L55

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+type CheckDetailsInfoProps = {
+    checkDetails?: ComplianceCheckResult | ComplianceClusterCheckStatus;
+    isLoading: boolean;
+    error?: Error;
+};
+
+function CheckDetailsInfo({ checkDetails, isLoading, error }: CheckDetailsInfoProps) {
+    if (error) {
+        return <div>Error: {error.message}</div>;
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx:28]: Can we render a warning alert here?

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                        {instructions}
+                    </DescriptionListDescription>
+                </DescriptionListGroup>
+                <DescriptionListGroup>
+                    <DescriptionListTerm>Labels</DescriptionListTerm>
+                    <DescriptionListDescription className="formatted-text">
+                        <List isPlain>
+                            {Object.entries(labels).map(([key, value]) => (
+                                <ListItem key={key}>
+                                    {key}: {value}
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx:68]: A question, not necessarily a suggestion, whether there is any convention in Kubernetes or OpenShift to render key in bold font weight?

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx:68]: CheckDetailsInfo

**Review by pedrottimark — APPROVED** [review]

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx (hunk 2 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx:28]: rushing 🤦 

Updating.

File: ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx (hunk 3 above)
Comment (bradr5) [reply ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/components/CheckDetailsInfo.tsx:68]: From the few examples I see in OpenShift, they don't seem to bold the key. However, there aren't really examples of labels being listed out like this. Every where that I see a label, it's in... well... a `Label`. Maybe that's a better solution.
Screenshot 2024-06-11 at 1 57 21 PM

**Pull request merged**